clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java

```diff
@@ -345,6 +345,9 @@ public void onFailure(RuntimeException e) {
                 cb.onComplete(offsets, e);
             }
         });
+
+        // ensure commit has a chance to be transmitted (without blocking on its completion)
+        client.poll(0);
```

Comment:
I debated on that. We allow WakeupException to be thrown from commitSync(), but maybe it's a little surprising to see it in commitAsync(). The other aspect of quickPoll() is that it doesn't execute scheduled tasks, so no heartbeats would be sent. But maybe we could expose task execution in an argument to quickPoll().